elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -118,59 +132,108 @@ public void readAndValidateConfigs() throws IOException {
     }
 
     /**
-     * Read security config file and checks for any missing Handlebar variables.
-     * @return boolean true if security config file exists else false
-     * @throws IOException
+     * Read and validates security config file.
      */
-    private void readSecurityConfig() throws IOException {
-        String key = Config.SECURITY.getConfigPath();
-        if (this.resourceMap.containsKey(key)) {
-            String content = IOUtils.toString(this.resourceMap.get(key).getInputStream(), StandardCharsets.UTF_8);
-            validateConfigForMissingVariables(content, this.variables);
-            this.setElideSecurityConfig(DynamicConfigHelpers.stringToElideSecurityPojo(content, this.variables));
-            validateRoleInSecurityConfig(this.getElideSecurityConfig());
-            this.resourceMap.remove(Config.SECURITY.getConfigPath());
-        }
+    private ElideSecurityConfig readSecurityConfig() {
+
+        return this.resourceMap
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> entry.getKey().startsWith(Config.SECURITY.getConfigPath()))
+                        .map(entry -> {
+                            try {
+                                String content = IOUtils.toString(entry.getValue().getInputStream(), UTF_8);
+                                validateConfigForMissingVariables(content, this.modelVariables);
+                                return DynamicConfigHelpers.stringToElideSecurityPojo(content, this.modelVariables);
+                            } catch (IOException e) {
+                                throw new IllegalStateException(e);
+                            }
+                        })
+                        .findAny()
+                        .orElse(new ElideSecurityConfig());
     }
 
     /**
-     * Read table config files and checks for any missing Handlebar variables.
-     * @throws IOException
+     * Read and validates db config files.
+     * @return Set<DBConfig> Set of SQL DB Configs
      */
-    private void readTableConfig() throws IOException {
-        Set<Table> tables = new HashSet<>();
-        if (this.resourceMap.isEmpty()) {
-            throw new IllegalStateException("No Table configs found at: " + this.configDir);
-        }
-        for (Entry<String, Resource> entry : this.resourceMap.entrySet()) {
-            String content = IOUtils.toString(entry.getValue().getInputStream(), StandardCharsets.UTF_8);
-            validateConfigForMissingVariables(content, this.variables);
-            ElideTableConfig table = DynamicConfigHelpers.stringToElideTablePojo(content, this.variables);
-            tables.addAll(table.getTables());
+    private Set<DBConfig> readDbConfig(Config config) {
+
+        return this.resourceMap
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> entry.getKey().startsWith(config.getConfigPath())).map(entry -> {
```

Comment:
map should go on its own line separate from filter.

Reply:
updated